akka-docs/src/main/paradox/stream/stages-overview.md

```diff
@@ -1326,10 +1326,28 @@ aggregated to the batched value.
 
 ---------------------------------------------------------------
 
+### extrapolate
+
+Allow for a faster downstream by expanding the last emitted element to an `Iterator`. For example, an
+`Iterator.continually(element)` will cause `extrapolate` to keep repeating the last emitted element. 
+
+All original elements are always emitted unchanged - the `Iterator` is only used whenever downstream signals demand, and
+there are no new elements from the upstream.
+
+Includes an optional `initial` argument to prevent blocking the entire stream when there are multiple producers.
```

Comment:
Isn't this rather, "when there is downstream demand before upstream emitted any element"?

Reply:
That's what the last part of the sentence is meant to convey ("and there are no new elements from the upstream").

But since you've missed that, it's probably unclear. I'll rewrite mostly per your suggestion.